### PR TITLE
service provider configuration docs

### PR DESCRIPTION
**Pull request by sparkoo**
### What does this PR do?
update docs with new oauth urls and with info how to configure service providers

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-250

### How to test this PR?

read and follow the doc

### CODE REVIEW

File: docs/ADMIN.md [hunk 1]
```diff
@@ -105,6 +104,16 @@ are also applicable to the oauth service. The configmap for oauth-service-specif
...
 
+## [Configuring Service Providers](#configuring-service-providers)
+
+OAuth requires to create OAuth Application on Service Provider side. Service providers usually require to set:
+ - __application homepage URL__: URL of the root of main Route/Ingress to SPI OAuth Service. For minikube: `https://spi.<minikube_ip>.nip.io`, for openshift: `https://<name>-<namespace>.<cluster_host>`. Example for minikube: `https://spi.192.168.64.166.nip.io`.
+ - __application callback URL__: homepage URL + `/oauth/callback`. Example for minikube: `https://spi.192.168.64.166.nip.io/oauth/callback`.
+
+_To create OAuth application at:_
+ - __GitHub__, follow [GitHub - Creating an OAuth App](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app)
+ - __GitLab__, follow [Configure GitLab as an OAuth 2.0 authentication identity provider](https://docs.gitlab.com/ee/integration/oauth_provider.html)
```
[the remark this answers is not in the document]
Comment (sparkoo) [reply docs/ADMIN.md:115]: I agree, but I couldn't find any relevant documentation for quay

**Review by metlos — APPROVED** [review]

File: oauth/handlers_test.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func TestCallbackSuccessHandler(t *testing.T) {
 func TestCallbackErrorHandler(t *testing.T) {
 	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
 	// pass 'nil' as the third parameter.
-	req, err := http.NewRequest("GET", "/github/callback?error=foo&error_description=bar", nil)
+	req, err := http.NewRequest("GET", "/oauth/callback?error=foo&error_description=bar", nil)
```
[the remark this answers is not in the document]
Comment (sparkoo) [reply oauth/handlers_test.go]: yes

File: oauth/handlers_test.go [hunk 3]
```diff
@@ -444,7 +444,7 @@ func TestMiddlewareHandlerCorsPart(t *testing.T) {
 func TestMiddlewareHandlerCors(t *testing.T) {
 	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
 	// pass 'nil' as the third parameter.
-	req, err := http.NewRequest("OPTIONS", "/github/authenticate?state=eyJhbGciO", nil)
+	req, err := http.NewRequest("OPTIONS", "/oauth/authenticate?state=eyJhbGciO", nil)
```
[the remark this answers is not in the document]
Comment (sparkoo) [reply oauth/handlers_test.go]: yes

File: docs/ADMIN.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sparkoo) [reply docs/ADMIN.md:115]: I've created an issue https://issues.redhat.com/browse/SVPI-275

**Pull request merged**